src/test/java/com/feedzai/commons/sql/abstraction/engine/testconfig/DatabaseTestUtil.java

```diff
@@ -66,25 +67,17 @@
 
     /**
      * Filters the instances given by {@code vendor}.
-     * <p/>
-     * The match is performed by contains, i.e. oracle matches oracle11 and oracle12.
+     * <p>
+     * The match is performed by the start of vendor name, i.e. h2 matches h2, h2remote, h2v2 and h2v2remote.
      *
      * @param vendors The list of instances to test.
      * @return The filtered instances to test.
      * @throws Exception If something wrong occurs reading the configurations.
      */
     public static Collection<DatabaseConfiguration> loadConfigurations(final String... vendors) throws Exception {
         return loadConfigurations().stream()
-            .filter(dbConfig -> {
-                for (String vendor : vendors) {
-                    if (dbConfig.vendor.contains(vendor)) {
-                        return true;
-                    }
-                }
-
-                return false;
-            })
-            .collect(Collectors.toList());
+                .filter(dbConfig -> Arrays.stream(vendors).anyMatch(dbConfig.vendor::startsWith))
+                .collect(Collectors.toList());
```

Comment:
Nitty-gritty detail

This will be called several times(I think), why not load the configurations only once with a `Suppliers#memoize`?
Also you could use `Predicates.or` instead creating a stream to iterate over the array for each loaded configuration.